.github/workflows/golangci-lint.yml

```diff
@@ -0,0 +1,45 @@
+name: golangci-lint
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+permissions:
+  contents: read
+
+jobs:
+  generate-matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - uses: actions/checkout@v4
+      - id: set-matrix
+        run: |
+          find . -type f -name 'go.mod' -printf '%h\n' \
+          | jq -r -R -s 'split("\n")[:-1] | sort | "matrix=\(.)"' >> $GITHUB_OUTPUT
+
+  golangci:
+    needs: generate-matrix
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        modules: ${{fromJson(needs.generate-matrix.outputs.matrix)}}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v4
+        with:
+          go-version-file: go.mod
+          # setup-go v4 uses cache automatically, which conflicts with golangci-lint's cache.
+          # See https://github.com/golangci/golangci-lint-action/pull/704
+          cache: false
+      # Work around missing go:embed file which causes a typecheck error.
+      # https://github.com/golangci/golangci-lint/issues/2912
+      - run: touch test/integration/versionchecker/testdata/test_manifests.tar
+      - uses: golangci/golangci-lint-action@v3
+        with:
+          version: v1.55.2
+          args: --timeout=30m --config=${GITHUB_WORKSPACE}/.golangci.ci.yaml
```

Comment:
Added an absolute path to the config file, so that it can be found from the modified working-directory below